src/Roslyn.Diagnostics.Analyzers/Core/ExportedPartsShouldHaveImportingConstructorCodeFixProvider.cs

```diff
@@ -0,0 +1,189 @@
+                if (!diagnostic.Properties.TryGetValue(nameof(ExportedPartsShouldHaveImportingConstructor.Scenario), out var scenario))
+                {
+                    continue;
+                }
+
+                string title;
+                Func<CancellationToken, Task<Document>> createChangedDocument;
+                switch (scenario)
+                {
+                    case ExportedPartsShouldHaveImportingConstructor.Scenario.ImplicitConstructor:
+                        title = RoslynDiagnosticsAnalyzersResources.ExportedPartsShouldHaveImportingConstructorCodeFix_ImplicitConstructor;
+                        createChangedDocument = cancellationToken => AddExplicitImportingConstructorAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    case ExportedPartsShouldHaveImportingConstructor.Scenario.NonPublicConstructor:
+                        title = RoslynDiagnosticsAnalyzersResources.ExportedPartsShouldHaveImportingConstructorCodeFix_NonPublicConstructor;
+                        createChangedDocument = cancellationToken => MakeConstructorPublicAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    case ExportedPartsShouldHaveImportingConstructor.Scenario.MissingAttribute:
+                        title = RoslynDiagnosticsAnalyzersResources.ExportedPartsShouldHaveImportingConstructorCodeFix_MissingAttribute;
+                        createChangedDocument = cancellationToken => AddImportingConstructorAttributeAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken);
+                        break;
+
+                    case ExportedPartsShouldHaveImportingConstructor.Scenario.MultipleConstructors:
+                    default:
+                        continue;
+                }
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title,
+                        createChangedDocument,
+                        equivalenceKey: scenario),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private async Task<Document> AddExplicitImportingConstructorAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var exportAttribute = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
+            var exportAttributeSymbol = semanticModel.GetSymbolInfo(exportAttribute, cancellationToken).Symbol?.ContainingType;
+            var importingConstructorAttributeSymbol = exportAttributeSymbol.ContainingNamespace.GetTypeMembers(nameof(ImportingConstructorAttribute)).FirstOrDefault();
```

Comment:
`exportAttributeSymbol` can be null based on the above conditional access `.Symbol?.ContainingType`.

Reply:
Fixing